vault/resource_consul_secret_backend_test.go

```diff
@@ -24,87 +30,132 @@ func TestConsulSecretBackend(t *testing.T) {
+					resource.TestCheckResourceAttr(resourceName, consts.FieldLocal, "false"),
+					resource.TestCheckResourceAttr(resourceName, "ca_cert", "FAKE-CERT-MATERIAL"),
+					resource.TestCheckResourceAttr(resourceName, "client_cert", "FAKE-CLIENT-CERT-MATERIAL"),
+					resource.TestCheckResourceAttr(resourceName, "client_key", "FAKE-CLIENT-CERT-KEY-MATERIAL"),
 				),
 			},
 			{
 				Config: testConsulSecretBackend_updateConfig_updateCerts(path, token),
 				Check: resource.ComposeTestCheckFunc(
-					resource.TestCheckResourceAttr("vault_consul_secret_backend.test", "path", path),
-					resource.TestCheckResourceAttr("vault_consul_secret_backend.test", "description", "test description"),
-					resource.TestCheckResourceAttr("vault_consul_secret_backend.test", "default_lease_ttl_seconds", "0"),
-					resource.TestCheckResourceAttr("vault_consul_secret_backend.test", "max_lease_ttl_seconds", "0"),
-					resource.TestCheckResourceAttr("vault_consul_secret_backend.test", "address", "consul.domain.tld:8501"),
-					resource.TestCheckResourceAttr("vault_consul_secret_backend.test", "token", token),
-					resource.TestCheckResourceAttr("vault_consul_secret_backend.test", "scheme", "https"),
-					resource.TestCheckResourceAttr("vault_consul_secret_backend.test", "local", "false"),
-					resource.TestCheckResourceAttr("vault_consul_secret_backend.test", "ca_cert", "FAKE-CERT-MATERIAL"),
-					resource.TestCheckResourceAttr("vault_consul_secret_backend.test", "client_cert", "UPDATED-FAKE-CLIENT-CERT-MATERIAL"),
-					resource.TestCheckResourceAttr("vault_consul_secret_backend.test", "client_key", "UPDATED-FAKE-CLIENT-CERT-KEY-MATERIAL"),
+					resource.TestCheckResourceAttr(resourceName, consts.FieldPath, path),
+					resource.TestCheckResourceAttr(resourceName, consts.FieldDescription, "test description"),
+					resource.TestCheckResourceAttr(resourceName, "default_lease_ttl_seconds", "0"),
+					resource.TestCheckResourceAttr(resourceName, "max_lease_ttl_seconds", "0"),
+					resource.TestCheckResourceAttr(resourceName, "address", "consul.domain.tld:8501"),
+					resource.TestCheckResourceAttr(resourceName, "token", token),
+					resource.TestCheckResourceAttr(resourceName, "scheme", "https"),
+					resource.TestCheckResourceAttr(resourceName, consts.FieldLocal, "false"),
+					resource.TestCheckResourceAttr(resourceName, "ca_cert", "FAKE-CERT-MATERIAL"),
+					resource.TestCheckResourceAttr(resourceName, "client_cert", "UPDATED-FAKE-CLIENT-CERT-MATERIAL"),
+					resource.TestCheckResourceAttr(resourceName, "client_key", "UPDATED-FAKE-CLIENT-CERT-KEY-MATERIAL"),
 				),
 			},
 		},
 	})
 }
 
+func TestConsulSecretBackend_Bootstrap(t *testing.T) {
+	backend := acctest.RandomWithPrefix("tf-test-backend")
+	resourceName := "vault_consul_secret_backend.test"
+	resourceRoleName := "vault_consul_secret_backend_role.test"
+
+	if testutil.CheckTestVaultVersion(t, "1.11") {
+		var consulAddr string
+		if envAddr := os.Getenv("CONSUL_ADDR"); envAddr != "" {
```

Comment:
I think we can drop this conditional in favor of https://github.com/hashicorp/vault/blob/ca804e5596e87b732eb7321b3d624d1883b29c5f/helper/testhelpers/consul/consulhelper.go#L33

We will probably just need to set the `CONSUL_HTTP_ADDR` env var instead.

Reply:
Oh nice catch! That's a neat, cleaner way to do it

Reply:
@benashz It looks like Consul doesn't like this naming format. Is it okay to use 127.0.0.1 in this case?
CI error here: https://github.com/hashicorp/terraform-provider-vault/runs/7660137651?check_suite_focus=true#step:4:692

Reply:
I think the issue is that `CONSUL_HTTP_ADDR` includes the scheme, if you remove that then it should work fine.

Reply:
It looks like the scheme should be included: https://www.consul.io/commands#consul_http_addr

Reply:
🤦 Thank you for pointing it out